pkg/operator/operator.go

```diff
@@ -136,7 +134,7 @@ func NewOperator() (context.Context, *Operator) {
 		LeaderElection:                !options.FromContext(ctx).DisableLeaderElection,
 		LeaderElectionID:              "karpenter-leader-election",
 		LeaderElectionResourceLock:    resourcelock.LeasesResourceLock,
-		LeaderElectionNamespace:       system.Namespace(),
+		LeaderElectionNamespace:       os.Getenv("SYSTEM_NAMESPACE"),
```

Comment:
This isn't needed at all actually. Leader namespace automatically becomes the namespace that the pod is running in when running in-cluster.